openedx/core/operations.py

```diff
@@ -1,14 +1,30 @@
 import os
 import signal
 import tempfile
+import gc
 
 from datetime import datetime
 from meliae import scanner
 
 
 def dump_memory(signum, frame):
-    """Dump memory stats for the current process to a temp directory. Uses the meliae output format."""
-    scanner.dump_all_objects('{}/meliae.{}.{}.dump'.format(tempfile.gettempdir(), datetime.now().isoformat(), os.getpid()))
+    """
+    Dump memory stats for the current process to a temp directory.
+    Uses the meliae output format.
+    """
+
+    scanner.dump_all_objects('{}/meliae.{}.{}.pre-gc.dump'.format(
+        tempfile.gettempdir(), datetime.now().isoformat(), os.getpid()
+    )
+
+    # force garbarge collection
+    for gen in xrange(3):
+        gc.collect(gen)
+        scanner.dump_all_objects(
+            '{}/meliae.{}.{}.gc-gen-{}.dump'.format(
+                tempfile.gettempdir(), datetime.now().isoformat(), os.getpid(), gen
+            )
+        )
```

Comment:
You should probably capture the timestamp early, and use the same stamp for all of the dumps, so that it's easy to package them together.